File: bookstore_backend/src/main/java/com/bookstore/services/AuthService.java
package com.bookstore.services;

import com.bookstore.dto.auth.AuthResponse;
import com.bookstore.dto.auth.LoginRequest;
import com.bookstore.dto.auth.RegisterRequest;

public interface AuthService {
    public void register(RegisterRequest req);
    public AuthResponse login(LoginRequest req);
    public AuthResponse refreshToken(String refreshToken);
    public void logout(String refreshToken);
    public void verifyEmail(String token);
}

File: bookstore_frontend/src/features/auth/services/authService.ts
import { apiClient } from '../../../lib/apiClient';
import type { ApiResponse, AuthResponse, ForgotPasswordRequest, LoginRequest, RegisterRequest, ResetPasswordRequest, UserResponse } from '../../../types/api';

export const authService = {
  login: (credentials: LoginRequest) => apiClient.post<ApiResponse<AuthResponse>>('/api/auth/login', credentials),
  register: (credentials: RegisterRequest) => apiClient.post<ApiResponse<void>>('/api/auth/register', credentials),
  verifyEmail: (token: string) => apiClient.get<ApiResponse<void>>(`/api/auth/verify-email?token=${token}`),
  forgotPassword: (request: ForgotPasswordRequest) => apiClient.post<ApiResponse<void>>('/api/auth/forgot-password', request),
  resetPassword: (request: ResetPasswordRequest) => apiClient.post<ApiResponse<void>>('/api/auth/reset-password', request),
  getProfile: () => apiClient.get<ApiResponse<UserResponse>>('/api/user/me'),
  logout: (refreshToken?: string) => apiClient.post<ApiResponse<void>>('/api/user/logout', { refreshToken }),
};
